### PR TITLE
Track cauldron procs

### PR DESCRIPTION
**Pull request by AardWolf**
Compiles, need to test as a submission.

Still needs the hunt_details to be added.

DO NOT APPROVE

### CODE REVIEW

**Comment by AardWolf** [comment]
It has passed testing. Example messages it would have sent are here: https://discord.com/channels/275500976662773761/355474934601875457/935563704420937788

File: src/scripts/main.js [hunk 1]
```diff
@@ -942,6 +943,43 @@
...
+                    const resultPotion = data.match(potionRegex)[1].trim();
+                    if ("inventory" in hunt_response && resultPotion in hunt_response.inventory) {
+                        const potionName = hunt_response.inventory[resultPotion].name;
+                        const potionId = hunt_response.inventory[resultPotion].item_id;
+                        if (potionName && potionId) {
+                            const convertible = {
+                                id: 3304, // Boiling Cauldron Trap
+                                name: "Boiling Cauldron Trap",
+                                quantity: 1,
+                            };
```
Comment (tehhowch) [inline src/scripts/main.js:968]: We should have a single place where these "known" convertibles get stored and retrieved from.

**Review by tehhowch — COMMENTED** [review]
Probably less fun to figure out when the procs could have triggered but did not, but something to consider anyway.

File: src/scripts/main.js [hunk 2]
```diff
@@ -942,6 +943,43 @@
                     }, window.origin);
                 }
             }
+            else if (markup.render_data.entry_timestamp === live_ts && 
+                css_class.search(/alchemists_cookbook_base_bonus/) !== -1) {
+
+                if (markup.render_data.entry_timestamp === live_ts) {
```
[the remark this answers is not in the document]
Comment (AardWolf) [reply src/scripts/main.js:949]: ah yes! Added the logic later, never updated the inner.

File: src/scripts/main.js [hunk 3]
```diff
@@ -942,6 +943,43 @@
...
+                    more_details['alchemists_cookbook_base_bonus'] = true;
+                    if (debug_logging) {window.console.log({procs: more_details});}
+                }
+            }
+            else if (markup.render_data.entry_timestamp === live_ts && 
+                    css_class.search(/boiling_cauldron_trap_bonus/) !== -1) {
+                const data = markup.render_data.text;
+                const potionRegex = /item\.php\?item_type=(.*?)"/; 
+                if (potionRegex.test(data)) {
+                    const resultPotion = data.match(potionRegex)[1].trim();
```
[the remark this answers is not in the document]
Comment (AardWolf) [reply src/scripts/main.js:959]: `"My Boiling Cauldron Trap brewed 1 <a class=\"\" title=\"\" href=\"https:\/\/www.mousehuntgame.com\/item.php?item_type=gauntlet_potion_3\" onclick=\"hg.views.ItemView.show('gauntlet_potion_3'); return false;\">Gauntlet Potion Tier 3<\/a>!"`

`"My Boiling Cauldron Trap brewed 1 <a class=\"\" title=\"\" href=\"https:\/\/www.mousehuntgame.com\/item.php?item_type=runic_cheese_potion\" onclick=\"hg.views.ItemView.show('runic_cheese_potion'); return false;\">Runic Potion<\/a>!"`

`"My Boiling Cauldron Trap brewed 1 <a class=\"\" title=\"\" href=\"https:\/\/www.mousehuntgame.com\/item.php?item_type=string_undead_emmental_potion\" onclick=\"hg.views.ItemView.show('string_undead_emmental_potion'); return false;\">Undead String Emmental Potion<\/a>!"`

(I have more if wanted)

You are right, I had a different regex originally but when I realized I needed to get the item id I figured it was better to get the name from there and not worry about the base doubling and pluralizing.

File: src/scripts/main.js [hunk 4]
```diff
@@ -942,6 +943,43 @@
...
+                    if ("inventory" in hunt_response && resultPotion in hunt_response.inventory) {
+                        const potionName = hunt_response.inventory[resultPotion].name;
+                        const potionId = hunt_response.inventory[resultPotion].item_id;
```
[the remark this answers is not in the document]
Comment (AardWolf) [reply src/scripts/main.js:962]: Neat!

File: src/scripts/main.js (hunk 1 above)
Comment (AardWolf) [reply src/scripts/main.js:968]: Agree. This is another of those weird ones like Riftios but this one has an item id from mousehunt so I used that. I could have put `user.weapon_item_id` instead -- if we confirmed the right weapon was armed or whatever other suspenders we wanted to add to it.

**Comment by AardWolf** [comment]
Changes made based on feedback, and tested. Still working.

If we want to add the hunt_detail for false we could do it by checking if the weapon is armed and setting it to false. Or we can do an outer join when querying MHCT.

**Pull request merged**